### PR TITLE
DSS Sync Failure Alerts

### LINKED ISSUE #2102: Expose sync status in DSS health endpoint

### PR DESCRIPTION
**Pull request by amarjandu**
Alerts #data-store-eng slack channel for sync failures.

Resolves: https://github.com/HumanCellAtlas/data-store/issues/2102

### CODE REVIEW

File: terraform/modules/env-alerts/dss.tf [hunk 1]
```diff
@@ -80,3 +80,30 @@ EOF
...
+}
+EOF
+
+  alarm_actions = ["${data.aws_sns_topic.alarms.arn}"]
+  ok_actions    = ["${data.aws_sns_topic.alarms.arn}"]
+
+  dimensions {
+    StateMachineArn = "arn:aws:states:us-east-1:861229788715:stateMachine:dss-sync-sfn-${var.env}"
+  }
+  # Hard-Coded due to no data.aws_sfn support in TF.
```
Comment (xbrianh) [inline terraform/modules/env-alerts/dss.tf]: Can we move this up two lines and change to something like "Terraform does not support ARN lookup for AWS Step Functions"

**Review by xbrianh — APPROVED** [review]
See comment, but LGTM

File: terraform/modules/env-alerts/dss.tf [hunk 2]
```diff
@@ -80,3 +80,30 @@ EOF
...
+  "environment": "${var.env}",
+  "description": "Sync Step-Function Failure!"
+}
+EOF
+
+  alarm_actions = ["${data.aws_sns_topic.alarms.arn}"]
+  ok_actions    = ["${data.aws_sns_topic.alarms.arn}"]
+
+  dimensions {
+    StateMachineArn = "arn:aws:states:us-east-1:861229788715:stateMachine:dss-sync-sfn-${var.env}"
```
Comment (mweiden) [inline terraform/modules/env-alerts/dss.tf]: You have to namespace the account id and region in order for this to work in prod.

**Review by mweiden — CHANGES_REQUESTED** [review]

File: terraform/modules/env-alerts/dss.tf (hunk 1 above)
Comment (amarjandu) [reply terraform/modules/env-alerts/dss.tf]: Yes, changed

File: terraform/modules/env-alerts/dss.tf (hunk 2 above)
Comment (amarjandu) [reply terraform/modules/env-alerts/dss.tf]: addressed; thanks for catching that.

**Comment by mweiden** [comment]
@amarjandu I fixed some errors in your terraform and pushed the change. Please merge this and apply the changes to each of your deployment environments. If you're not a `dcp-admin` you may need @xbrianh's help.

**Comment by amarjandu** [comment]
> @amarjandu I fixed some errors in your terraform and pushed the change. Please merge this and apply the changes to each of your deployment environments. If you're not a `dcp-admin` you may need @xbrianh's help.

thanks for the help @mweiden , i was looking at the posted ARN for the statemachine with the addition of Namespacing the Region/Account ID. [AWS Docs](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-step-functions)

I think we would want to add 
```
data "aws_region" "current" {}
``` 
to `env-alerts/main.tf` 

then format

StateMachineArn = "arn:aws:states:**${var.env}**:${data.aws_caller_identity.current.account_id}:stateMachine:dss-sync-sfn-${var.env}"
  }
 to 

StateMachineArn = "arn:aws:states:**${data.aws_region.current.name}**:${data.aws_caller_identity.current.account_id}:stateMachine:dss-sync-sfn-${var.env}"
  }

i have the changes committed, and can push it, then get @xbrianh to help me out.

**Comment by mweiden** [comment]
@amarjqndu in fogg, ‘var.env’ is correct. Please use that.

**Comment by amarjandu** [comment]
@xbrianh addressed `var.env` issue, changed to `var.region`.

**Review by mweiden — APPROVED** [review]

**Pull request merged**